src/NonlinearArithmetic/Logarithm.dfy

```diff
@@ -0,0 +1,103 @@
+// RUN: %dafny /compile:0 /noNLarith "%s"
+
+include "Mul.dfy"
+include "DivMod.dfy"
+include "Power.dfy"
+
+module Logarithm {
```

Comment:
Could you perhaps name this module/file `Integers`? That way, we would have `Integers.Log`

Reply:
This isn't how the rest of this library is structured; I tried to stay consistent with what was already there.
(Also, if you named it `Integers`, wouldn't you want to have other modules also named `Integers` in other files?)

Reply:
Mmmh that makes me think it should just be included in a Math module so that we have Math.log. Let's do that later.